be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -456,8 +487,52 @@ Status SegmentIterator::_get_row_ranges_by_keys() {
             _scan_range.add(Range{lower_rowid, upper_rowid});
         }
     }
-    _opts.stats->rows_key_range_filtered += num_rows() - _scan_range.span_size();
-    StarRocksMetrics::instance()->segment_rows_by_short_key.increment(_scan_range.span_size());
+
+    return Status::OK();
+}
+
+Status SegmentIterator::_get_row_ranges_by_short_key_ranges() {
+    DCHECK(!_opts.short_key_ranges.empty());
+    DCHECK_EQ(0, _scan_range.span_size());
+
+    if (_opts.short_key_ranges.size() == 1 && _opts.short_key_ranges[0]->lower->is_infinite() &&
+        _opts.short_key_ranges[0]->upper->is_infinite()) {
+        _scan_range.add(Range(0, num_rows()));
+        return Status::OK();
+    }
+
+    RETURN_IF_ERROR(_segment->load_index(StorageEngine::instance()->tablet_meta_mem_tracker()));
+    for (const auto& short_key_range : _opts.short_key_ranges) {
+        rowid_t lower_rowid = 0;
+        rowid_t upper_rowid = num_rows();
+
+        const auto& upper = short_key_range->upper;
+        if (upper->tuple_key != nullptr) {
+            _init_column_iterators<false>(upper->tuple_key->schema());
```

Comment:
check return status?

Reply:
It is the original code before this PR.
Before `_get_row_ranges_by_keys`, `RETURN_IF_ERROR(_init_column_iterators<true>(_schema))` has been called. Therefore, here it is guaranteed that the result must be `OK`.